### ISSUE TITLE
Добавить статью про веб-компоненты

### ISSUE THREAD

**Issue by notiv-nt** [doka-guide/content#2068]

**Comment by pepelsbey** [comment]
Сначала стоит определиться, куда пойдёт вводна статья про веб-компоненты. Кажется, что в JS, ведь для их инициализации нужен JS.

**Comment by nlopin** [comment]
> Сначала стоит определиться, куда пойдёт вводна статья про веб-компоненты. Кажется, что в JS, ведь для их инициализации нужен JS.

Можем целый подраздел создать, если захотим развить тему

**Comment by nlopin** [comment]
Чем больше смотрю на список, тем больше мне нравится идея отдельного раздела, но он тогда будет очень маленьким. Думаю стоит начать со статей по списку и большой обзорной статьи, коротая будет связывать все разроненные кусочки вместе.

Обзорную статью зафичерить и посмотреть на интерес публики и решить по отдельному разделу после.

@pepelsbey @solarrust как вам?

**Comment by web-padawan** [comment]
Я могу начать с отдельных статей в порядке: `attachShadow()` - `:host` - `<slot>` - `::slotted()` - и тд.

Тема Shadow DOM в целом сложнее и с ней хорошо бы определиться первой. Custom elements проще.
Также, экспериментальные фичи стоит отдельно описать (Form Participation API, custom state pseudo-classes).

Есть моменты, которые я бы не документировал, поскольку Safari возражает против их реализации в текущем варианте спеки (атрибут `is=""` и customised built-in elements) или черновика (Constructable Stylesheets).

**Comment by pepelsbey** [comment]
> Обзорную статью зафичерить и посмотреть на интерес публики и решить по отдельному разделу после.

Мы сейчас про раздел или категорию? Мне кажется, что совсем раздел (уровня HTML, CSS, JS) делать не стоит, пока у нас нет видения как и когда создавать новые. Особенно для кросстехнологического стека, вроде веб-компонентов.

**Comment by web-padawan** [comment]
Согласен с Вадимом, необходимости в разделе не вижу. Обзорная статья может быть и в JS, в то же время отдельная статья именно про кастомные элементы болеее уместна в HTML, на мой взгляд.

**Comment by nlopin** [comment]
Да, я предлагаю то же самое. Сначала материал в те разделы, куда они и принадлежат.

**Comment by DrakesBot12** [comment]
Ещё актуально? И я не до конца понял, нужно сначало статьи описанные @web-padawan сделать а только потом статью о Веб компонентах делать? И в каком раделе будет эта статья?

**Comment by DrakesBot12** [comment]
Я могу в принципе сделать драфт-пиар и туда потихоньку добавлять статьи описанные @web-padawan

**Comment by solarrust** [comment]
Нет какого-то одного плана. Ты можешь писать материалы в любом удобном для тебя порядке. А раздел надо будет выбирать для каждой отдельной статьи, куда она больше подойдёт

**Comment by DrakesBot12** [comment]
Угу, то есть допустим я напишу статью про веб-компоненты и потом после её написания буду в пиар этот добавлять статьи допустим про slot, :host() и т.п., норм будет?

**Comment by DrakesBot12** [comment]
Я всё же решаюсь написать про веб-компоненты :)
